packages/component-library/components/Heading/Heading.elm

```diff
@@ -0,0 +1,270 @@
+module Heading.Heading exposing
+    ( Config
+    , DataAttribute
+    , TypeVariant(..)
+    , a
+    , addAttribute
+    , classNameAndIHaveSpokenToDST
+    , div
+    , h1
+    , h2
+    , h3
+    , h4
+    , h5
+    , h6
+    , id
+    , p
+    , pre
+    , span
+    , tag
+    , variant
+    , view
+    )
+
+import CssModules exposing (css)
+import Debug exposing (log)
+import Html exposing (text)
+import Html.Attributes
+import List
+
+
+type Config msg
+    = Config (ConfigValue msg)
+
+
+type alias Element msg =
+    List (Html.Attribute msg) -> List (Html.Html msg) -> Html.Html msg
+
+
+type alias DataAttribute =
+    { name : String
+    , value : String
+    }
+
+
+type alias ConfigValue msg =
+    { tag : Maybe (Element msg)
+    , variant : TypeVariant
+    , id : Maybe String
+    , data : List DataAttribute
+    , classNameAndIHaveSpokenToDST : Maybe String
+    }
+
+
+defaultConfig : ConfigValue msg
+defaultConfig =
+    { tag = Nothing
+    , variant = Display0
+    , id = Nothing
+    , data = []
```

Comment:
I think in my head, rather than a `data` properties, there would be an `attributes` property, that could contain many different types of attributes: data, aria attributes (eventually). It's find if it's just data attributes for now, but some of the other names should reflect this. (wait, I'll link a comment)

Reply:
I'll change `data` to `dataAttribute` for clarity